cli/cli/out/hook.go

```diff
@@ -0,0 +1,28 @@
+package out
+
+import (
+	"io"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Hook is a hook that writes logs of specified LogLevels to specified Writer
+type Hook struct {
+	Writer    io.Writer
+	LogLevels []log.Level
+	Formatter log.Formatter
+}
+
+func (hook *Hook) Fire(entry *log.Entry) error {
+	line, err := hook.Formatter.Format(entry)
+	if err != nil {
+		return err
```

Comment:
could we add stacktrace here ?

Reply:
@leoporoli thinking about this --> do you think it would be better to maybe silently fail here? The pros is that users won't be errored out because of logging issue. The downside would be that we would have no way of knowing if it's failing. One way to get around is hook a metric with it - which we can follow.

Reply:
or; i can create another logger without hooks - which i can use to print warnings -- and return error as nil.

Reply:
ok; let me silently fail here atm - add the extra capabilities in next prs. @leoporoli